### PR TITLE
gh-104635: Eliminate redundant STORE_FAST instructions in the compiler

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-104635

## Benchmarks

```
import pyperf

runner = pyperf.Runner()
runner.timeit(name="bench dead_store",
              stmt="""
_, _, _, a = 1, 2, 3, 4
""")

Mean +- std dev: [base] 11.7 ns +- 0.2 ns -> [opt] 11.1 ns +- 0.1 ns: 1.05x faster
```

```
import pyperf

runner = pyperf.Runner()
runner.timeit(name="bench static swaps",
              stmt="""
_ = f(True, False)
""",
setup="""
def f(x, y):
    a, a = x, y
    return a
""")


Mean +- std dev: [static_base] 27.1 ns +- 0.3 ns -> [static_opt] 26.0 ns +- 0.2 ns: 1.04x faster
```

### CODE REVIEW

**Comment by corona10** [comment]
@carljm cc @iritkatriel 
 
Mark's change + dead_store elimination + apply_static_swaps makes 3-4% faster without adding super instruction.

File: Lib/test/test_peepholer.py [hunk 1]
```diff
@@ -1075,15 +1075,56 @@ def test_no_unsafe_static_swap(self):
             ('RETURN_VALUE', 5)
         ]
         expected_insts = [
+            ('LOAD_CONST', 0, 1),
+            ('LOAD_CONST', 1, 2),
+            ('NOP', 0, 3),
+            ('STORE_FAST', 1, 4),
+            ('POP_TOP', 0, 4),
+            ('RETURN_VALUE', 5)
+        ]
```
Comment (corona10) [inline Lib/test/test_peepholer.py:1084]: 8 instructions -> 5 instructions

File: Lib/test/test_peepholer.py [hunk 2]
```diff
@@ -1075,15 +1075,56 @@ def test_no_unsafe_static_swap(self):
...
+        expected_insts = [
+            ('LOAD_CONST', 0, 1),
+            ('LOAD_CONST', 1, 2),
+            ('NOP', 0, 3),
             ('POP_TOP', 0, 4),
+            ('STORE_FAST', 1, 4),
+            ('RETURN_VALUE', 5)
+        ]
```
Comment (corona10) [inline Lib/test/test_peepholer.py:1104]: 7 instructions -> 5 instructions

**Comment by corona10** [comment]
@carljm Can you please take a look at?

**Review by carljm — APPROVED** [review]
LGTM!

**Pull request merged**